packages/running-extension/src/kernels.tsx

```diff
@@ -87,7 +157,14 @@ export async function addKernelRunningSessionManager(
     shutdownAllLabel: trans.__('Shut Down All'),
     shutdownAllConfirmationText: trans.__(
       'Are you sure you want to permanently shut down all running kernels?'
-    )
+    ),
+    toolbarButtons: [
+      new CommandToolbarButton({
+        commands,
+        id: SHUTDOWN_UNUSED_BUTTON_CLASS,
```

Comment:
From a quick glance the logic looks all good now. The last thing I would suggest is using an icon instead of text for this button. Let's first choose and add an icon to the command. I think some kind of material design icon for cleaning could do?

```suggestion
        icon: someIcon,
        id: SHUTDOWN_UNUSED_BUTTON_CLASS,
```

Reply:
![image](https://github.com/jupyterlab/jupyterlab/assets/46305192/a50de9c7-8959-4aee-a8bf-9ea295741472)

I added one, I don't see very many options
I manually added it to `/jupyterlab/packages/ui-components/src/icon/iconimports.ts` which needs to be fixed since it's supposed to be auto-generated

Should the text for `Shut Down All` and `Shut Down Unused` be removed, and just have icons instead, since there's not that much room